src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/InteropServices/CSharpDynamicInterfaceCastableImplementation.Fixer.cs

```diff
@@ -0,0 +1,255 @@
+            List<SyntaxNode> generatedMembers = new List<SyntaxNode>();
+            foreach (var iface in type.AllInterfaces)
+            {
+                foreach (var member in iface.GetMembers())
+                {
+                    if (!member.IsStatic && type.FindImplementationForInterfaceMember(member) is null)
+                    {
+                        SyntaxNode? implementation = member.Kind switch
+                        {
+                            SymbolKind.Method => GenerateMethodImplementation((IMethodSymbol)member),
+                            SymbolKind.Property => GeneratePropertyImplementation((IPropertySymbol)member),
+                            SymbolKind.Event => GenerateEventImplementation((IEventSymbol)member, declaration, generator, defaultMethodBodyStatements),
+                            _ => null
+                        };
+                        if (implementation is not null)
+                        {
+                            generatedMembers.Add(generator.AsPrivateInterfaceImplementation(
+                                implementation,
+                                generator.NameExpression(member.ContainingType)));
+                        }
+                    }
+                }
+            }
+
+            // Explicitly use the C# syntax APIs to work around https://github.com/dotnet/roslyn/issues/53605
+            TypeDeclarationSyntax typeDeclaration = (TypeDeclarationSyntax)declaration;
+            typeDeclaration = typeDeclaration.AddMembers(generatedMembers.Cast<MemberDeclarationSyntax>().ToArray());
+
+            editor.ReplaceNode(declaration, typeDeclaration);
+
+            return editor.GetChangedDocument();
+
+            SyntaxNode? GenerateMethodImplementation(IMethodSymbol method)
+            {
+                // Method with associated symbols are not standalone methods that a user would write in C#.
+                // They're methods like property or event accessors, which are covered in the following local functions.
+                if (method.AssociatedSymbol is not null)
+                {
+                    return null;
+                }
+                SyntaxNode methodDecl = generator.MethodDeclaration(method);
+                methodDecl = generator.WithModifiers(methodDecl, generator.GetModifiers(declaration).WithIsAbstract(false));
+                return generator.WithStatements(methodDecl, defaultMethodBodyStatements);
+            }
+
+            SyntaxNode GeneratePropertyImplementation(IPropertySymbol property)
+            {
+                SyntaxNode propertyDecl = property.IsIndexer
+                    ? generator.IndexerDeclaration(property)
+                    : generator.PropertyDeclaration(property);
+                propertyDecl = generator.WithModifiers(propertyDecl, generator.GetModifiers(declaration).WithIsAbstract(false));
```

Comment:
No. The implementations must be non-virtual, otherwise the runtime will attempt interface dispatch and fail.

Reply:
Do you have a test case covering this? Might be easier for me to check what this is covering so I understand fully

Reply:
The `DynamicInterfaceCastableImplementation_NoPropertiesImplementated_CS_Diagnostic` test should cover this scenario.

Reply:
`DynamicInterfaceCastableImplementation_NoPropertiesImplementated_CS_Diagnostic ` doesn't seem to have anything abstract. Is it even possible for the declaration to be abstract in this case? I guess that's my bigger question. The code suggests yes, but I'm not sure why. It's technically right, but if there's a reason `declaration` could be abstract here a comment is probably warranted as to why.

Reply:
The property declaration in an interface is implicitly abstract and (as part of Default Interface Members) it can be explicitly declared abstract as well (see https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQGYACLEgYRIG8CT6zT0AWEgWQAoBKGuhgXwKD8BWABcApgCcAZhADGEkgEkCtfAxIRgAZzFSFYkuJIAxAPbmaAcwliA3Drv3h/IA).

I'll add some tests around this.

Reply:
I see. In this case abstract properties are the same as other properties on interface declarations?

Reply:
Yep. Any property without a body declared on an interface is considered `abstract`.